test/e2e/operator_test.go

```diff
@@ -1842,8 +1842,9 @@ func TestIngressControllerCustomEndpoints(t *testing.T) {
 		}
 		defer func() {
 			// Remove the custom endpoints from the infrastructure config.
-			infraConfig.Spec.PlatformSpec.AWS = nil
-			if err := kclient.Update(context.TODO(), &infraConfig); err != nil {
+			if err := updateInfrastructureConfigSpecWithRetryOnConflict(t, types.NamespacedName{Name: "cluster"}, timeout, func(spec *configv1.InfrastructureSpec) {
```

Comment:
Any reason not to use `updateInfrastructureConfigSpecWithRetryOnConflict` on line 1826 too?

Reply:
None. And I first did do that. Then removed the change as it wasn't strictly needed for the test to pas. I can add it back again because when you read through the test it does stand out as a bit weird to do largely the same thing but in two different ways.